packages/scaffolder-cli/src/commands/generate/scaffolderApi.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  ScaffolderDryRunOptions,
+  ScaffolderDryRunResponse,
+} from '@backstage/plugin-scaffolder-react';
+
+const scaffolderApi = {
+  dryRun: async (
+    options: ScaffolderDryRunOptions,
+  ): Promise<ScaffolderDryRunResponse> => {
+    const response = await fetch(
+      'http://localhost:7007/api/scaffolder/v2/dry-run', // TODO: make port configurable
```

Comment:
I know this is a Draft but just wanted to mention that this won't work as Backstage is secure by default since `1.24.0` which means all the APIs need to be called with a proper token.

Reply:
aaa, thank you @awanlin ! 😄 This is exactly why i made a draft so I wouldn't go off in the wrong direction. 👌 
This works for me locally with the `example-backend` but maybe that's a special case?

Reply:
Oh, another thing to consider here is that hardcoding `localhost` would be limiting for many people as they might want to point this at a different environment for various reasons.